pkg/serializers/sneakers/serializer.go

```diff
@@ -0,0 +1,69 @@
+package sneakers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/benmanns/goworker/pkg/interfaces"
+)
+
+// JSONSerializer implements the Serializer interface for JSON
+type JSONSerializer struct {
+	useNumber bool
+}
+
+// NewSerializer creates a new JSON serializer
+func NewSerializer() *JSONSerializer {
+	return &JSONSerializer{
+		useNumber: false,
+	}
+}
+
+// Serialize converts a job to JSON bytes
+func (s *JSONSerializer) Serialize(job interfaces.Job) ([]byte, error) {
+	message := ConstructMessage(job)
+
+	data, err := json.Marshal(message)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal job: %w", err)
+	}
+
+	return data, nil
+}
+
+// Deserialize converts JSON bytes to a job
+func (s *JSONSerializer) Deserialize(data []byte) (interfaces.Job, error) {
```

Comment:
I believe you'll have to update definition in serializer interface as well, currently it's:
```
Deserialize(data []byte, metadata JobMetadata) (Job, error)
```